e2e test run locally timing out

### What broke? What's expected?

When running the e2e tests locally they keep timing out. It can be at different stages of the [plugin v2 tests](https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v3/plugin_cluster_test.go). Output will be something similar to the following (depending on where it times out):

```console
$ make test-e2e-local

[...]

STEP: deploying the controller-manager
running: make deploy IMG=e2e-test/controller-manager:gkst
STEP: validating that the controller-manager pod is running as expected
running: kubectl -n e2e-gkst-system get pods -l control-plane=controller-manager -o go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}{{ "\n" }}{{ end }}{{ end }}
running: kubectl -n e2e-gkst-system get pods e2e-gkst-controller-manager-5c54dd976b-kkljl -o jsonpath={.status.phase}
running: kubectl -n e2e-gkst-system get pods -l control-plane=controller-manager -o go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}{{ "\n" }}{{ end }}{{ end }}
running: kubectl -n e2e-gkst-system get pods e2e-gkst-controller-manager-5c54dd976b-kkljl -o jsonpath={.status.phase}
running: kubectl -n e2e-gkst-system get pods -l control-plane=controller-manager -o go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}{{ "\n" }}{{ end }}{{ end }}
running: kubectl -n e2e-gkst-system get pods e2e-gkst-controller-manager-5c54dd976b-kkljl -o jsonpath={.status.phase}
panic: test timed out after 20m0s

[...]
```

### Reproducing this issue

Run `make test-e2e-local` on your local setup.

### KubeBuilder (CLI) Version

main branch

### PROJECT version

_No response_

### Plugin versions

_No response_

### Other versions

_No response_

### Extra Labels

_No response_

--- a/test/e2e/local.sh
+++ b/test/e2e/local.sh
@@ -18,7 +18,7 @@ source "$(dirname "$0")/../common.sh"
 source "$(dirname "$0")/setup.sh"
 
 export KIND_CLUSTER="local-kubebuilder-e2e"
-create_cluster ${KIND_K8S_VERSION:-v1.18.0}
+create_cluster ${KIND_K8S_VERSION:-v1.18.15}
 if [ -z "${SKIP_KIND_CLEANUP:-}" ]; then
   trap delete_cluster EXIT
 fi

--- a/test/e2e/setup.sh
+++ b/test/e2e/setup.sh
@@ -51,5 +51,5 @@ function test_cluster {
   local flags="$@"
 
   go test $(dirname "$0")/v2 $flags
-  go test $(dirname "$0")/v3 $flags -timeout 20m
+  go test $(dirname "$0")/v3 $flags -timeout 30m
 }

